### PR TITLE
Support for PARTNER Ips

### PR DESCRIPTION
**Pull request by Ark2307**

### CODE REVIEW

File: apps/api-runtime/src/main/java/com/akto/runtime/policies/AktoPolicyNew.java [hunk 1]
```diff
@@ -181,7 +187,7 @@ public void process(HttpResponseParams httpResponseParams) throws Exception {
                     break;
                 case DETERMINE_API_ACCESS_TYPE:
                     try {
-                        saveSample = apiAccessTypePolicy.findApiAccessType(httpResponseParams, apiInfo, filter);
```
Comment (ayushaga14) [inline apps/api-runtime/src/main/java/com/akto/runtime/policies/AktoPolicyNew.java:184]: why?

File: apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java [hunk 2]
```diff
@@ -13,16 +14,18 @@
 
 public class ApiAccessTypePolicy {
     private List<String> privateCidrList;
-    public static final String X_FORWARDED_FOR = "x-forwarded-for";
+    private List<String> partnerIpsList;
+
+	public static final String X_FORWARDED_FOR = "x-forwarded-for";
```
Comment (ayushaga14) [inline apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java:19]: fix indent

File: apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java [hunk 3]
```diff
@@ -35,23 +38,30 @@ public boolean findApiAccessType(HttpResponseParams httpResponseParams, ApiInfo
...
+                        apiInfo.getApiAccessTypes().add(ApiAccessType.PUBLIC);
+                        return;
+                    }
                 }
            } catch (Exception e) {
-                return false;
+                return;
            }
         }
+        ApiAccessType accessType = isAccessTypePartner ? ApiAccessType.PARTNER : ApiAccessType.PRIVATE;
```
Comment (ayushaga14) [inline apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java:61]: can be done in if(isPartnerIp(ip)){ block

File: apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java [hunk 4]
```diff
@@ -13,16 +14,18 @@
...
     private static final Logger logger = LoggerFactory.getLogger(ApiAccessTypePolicy.class);
 
     public ApiAccessTypePolicy(List<String> privateCidrList) {
         this.privateCidrList = privateCidrList;
     }
 
 
-    public boolean findApiAccessType(HttpResponseParams httpResponseParams, ApiInfo apiInfo, RuntimeFilter filter) {
-        if (privateCidrList == null || privateCidrList.isEmpty()) return false;
+    public void findApiAccessType(HttpResponseParams httpResponseParams, ApiInfo apiInfo, RuntimeFilter filter) {
```
Comment (ayushaga14) [inline apps/api-runtime/src/main/java/com/akto/runtime/policies/ApiAccessTypePolicy.java:27]: why return type is removed?

File: apps/dashboard/src/main/java/com/akto/action/AdminSettingsAction.java [hunk 5]
```diff
@@ -31,8 +33,11 @@ public String execute() throws Exception {
 
     public AccountSettings.SetupType setupType;
     public Boolean newMergingEnabled;
+    private List<String> privateCidrList;
+
+	private List<String> partnerIpList;
```
Comment (ayushaga14) [inline apps/dashboard/src/main/java/com/akto/action/AdminSettingsAction.java:38]: indent

File: apps/dashboard/src/main/java/com/akto/action/AdminSettingsAction.java [hunk 6]
```diff
@@ -31,8 +33,11 @@ public String execute() throws Exception {
...
 
-    public String updateSetupType() {
+	public String updateSetupType() {
```
Comment (ayushaga14) [inline apps/dashboard/src/main/java/com/akto/action/AdminSettingsAction.java:40]: indent

**Review by ayushaga14 — CHANGES_REQUESTED** [review]
resolve comments

File: apps/api-runtime/src/main/java/com/akto/runtime/policies/AktoPolicyNew.java (hunk 1 above)
Comment (Ark2307) [reply apps/api-runtime/src/main/java/com/akto/runtime/policies/AktoPolicyNew.java:184]: fixed method, it was always returning false earlier, so turned into void, referred avneesh

**Pull request closed without merging**